encoding/da.go

```diff
@@ -19,15 +19,31 @@ const (
 	// CodecV1 represents the version 1 of the encoder and decoder.
 	CodecV1
 
+	// CodecV2 represents the version 2 of the encoder and decoder.
+	CodecV2
+
 	// txTypeTest is a special transaction type used in unit tests.
 	txTypeTest = 0xff
```

Comment:
It looks weird to have `txTypeTest` in the same "enum declaration" as the codec versions, can we separate these?

Reply:
Yeah. it's a hack way, only used in testing the challenge points with some "assigned payload", better remove it.

One way is to add another `constructBlobPayloadWithAssignedTxData`, adding another ConvertTxDataToRLPEncoding to assign the input directly as the rlp-encoding.

Or some smaller changes to achieve this?